lib/iris/fileformats/netcdf/loader.py

```diff
@@ -518,18 +520,18 @@ def load_cubes(filenames, callback=None, constraints=None):
     # Create an actions engine.
     engine = _actions_engine()
 
-    if isinstance(filenames, str):
-        filenames = [filenames]
+    if isinstance(file_sources, str) or not isinstance(file_sources, Iterable):
```

Comment:
What about paths?

Reply:
Update : **Works ok**
for Path / str / Dataset or any iterable of same.
 -- only "str" requires specially treatment "as if not an iterable" : Path is not iterable anyway.